src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -160,6 +165,8 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_variances(
 
 	// get kernel matrix and create eigen representation of it
 	SGMatrix<float64_t> k_tsts=kernel->get_kernel_matrix();
+	//To DO in order to speed up the prediction process,
+	//we only need to obtain the diagonal elements of the kernel matrix
```

Comment:
@karlnapf sorry man I didn't follow. If only diagonal is required, couldn't we have used here `CKernel::get_kernel_diagonal()` instead?